services/app-api/src/resolvers/contract/updateDraftContractRates.test.ts

```diff
@@ -0,0 +1,686 @@
+
+        const linkResult = await stateServer.executeOperation({
+            query: UPDATE_DRAFT_CONTRACT_RATES,
+            variables: {
+                input: {
+                    contractID: contractDraft.id,
+                    updatedRates: [
+                        {
+                            type: 'LINK',
+                            rateID: foreignRateID,
+                        },
+                    ],
+                },
+            },
+        })
+
+        expect(linkResult.errors).toBeUndefined()
+        if (!linkResult.data) {
+            throw new Error('no linkResult')
+        }
+
+        const draftRates =
+            linkResult.data.updateDraftContractRates.contract.draftRates
+
+        expect(draftRates).toHaveLength(1)
+
+        const unlinkResult = await stateServer.executeOperation({
+            query: UPDATE_DRAFT_CONTRACT_RATES,
+            variables: {
+                input: {
+                    contractID: contractDraft.id,
+                    updatedRates: [],
+                },
+            },
+        })
+
+        expect(unlinkResult.errors).toBeUndefined()
+        if (!unlinkResult.data) {
+            throw new Error('no unlinkResult')
+        }
+
+        const emptyDraftRates =
+            unlinkResult.data.updateDraftContractRates.contract.draftRates
+
+        expect(emptyDraftRates).toHaveLength(0)
+    })
+
+    it('deletes a draft rate', async () => {
+        const stateServer = await constructTestPostgresServer()
+
+        const draft = await createAndUpdateTestHealthPlanPackage(stateServer)
+        const draftFD = latestFormData(draft)
+        const rate = draftFD.rateInfos[0]
+
+        if (!rate.id) {
+            throw new Error('rate no id')
+        }
+
+        const existingRateResult = await fetchTestRateById(stateServer, rate.id)
+        expect(existingRateResult.id).toBeDefined()
+
+        const result = await stateServer.executeOperation({
+            query: UPDATE_DRAFT_CONTRACT_RATES,
+            variables: {
+                input: {
+                    contractID: draft.id,
+                    updatedRates: [],
+                },
+            },
+        })
+        expect(result.errors).toBeUndefined()
+
+        if (!result.data) {
+            throw new Error('no data returned')
+        }
+
+        const draftRates =
+            result.data.updateDraftContractRates.contract.draftRates
+
+        expect(draftRates).toHaveLength(0)
+
+        const nonexistantRateResult = await stateServer.executeOperation({
+            query: FETCH_RATE,
+            variables: { input: { rateID: rate.id } },
+        })
+
+        expect(nonexistantRateResult.errors).toBeDefined()
+        if (!nonexistantRateResult.errors) {
+            throw new Error('define errors')
+        }
+        expect(nonexistantRateResult.errors[0].extensions?.code).toBe(
+            'NOT_FOUND'
+        )
+    })
+
+    it.todo('doesnt allow updating a non-child')
```

Comment:
Where are we testing the error path for trying to link or delink rates that do not exist or are not yet submitted?  That feels like a possible error if the frontend was reading rate IDs (for example passing revision id) or handling API data incorrectly